runtime/src/bank.rs

```diff
@@ -1495,7 +1497,13 @@ impl Bank {
     pub fn withdraw(&self, pubkey: &Pubkey, lamports: u64) -> Result<()> {
         match self.get_account(pubkey) {
             Some(mut account) => {
-                if lamports > account.lamports {
+                let min_balance = match get_system_account_kind(&account) {
```

Comment:
this function has always been borken, it has never enforced that the account be system_program...  I don't think you need to fix it here for Nonce, maybe delete it and see what breaks (in another PR)?

Reply:
There was briefly some code here, similar to https://github.com/solana-labs/solana/blob/4c35fdd535def77c1971058eaf3e837e0c783b71/runtime/src/accounts.rs#L138-L141, but it broken a bunch of tests and I bailed. Chalked it up to out of scope.  I can definitely readdress in another PR.

I think we still need the `Nonce` check to protect the rent-exempt status though, no?

Reply:
I think `withdraw()` should probably die

Reply:
I can get behind that (in the future) 😉